mlflow/tensorflow.py

```diff
@@ -333,3 +343,125 @@ def predict(self, df):
             raw_preds = self.tf_sess.run(self.output_tensors, feed_dict=feed_dict)
             pred_dict = {column_name: values.ravel() for column_name, values in raw_preds.items()}
             return pandas.DataFrame(data=pred_dict)
+
+
+class __MLflowTfKerasCallback(Callback):
+    def __init__(self):
+        if mlflow.active_run() is None:
+            mlflow.start_run()
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def on_epoch_end(self, epoch, logs=None):
+        pass
+
+    def on_train_end(self, logs=None):
+        mlflow.log_param('learning_rate', tensorflow.keras.backend.eval(self.model.optimizer.optimizer._lr))
```

Comment:
Nice, is there any way we could save more info about the actual model structure? For example, maybe we can log `num_layers` (the number of model layers) as a param & `model.get_config()` (or maybe `model.print_summary()`) as an artifact? See https://keras.io/models/about-keras-models/

Reply:
Also, are all optimizers guaranteed to have `_lr` and `_epsilon` as fields? Is there a public API we could use to access those fields? Otherwise we might want to consider try-catching this. Also, we can use `mlflow.log_params(<dict>)` to log multiple param key-value pairs at once

Reply:
After some offline discussion, added additional parameter logging to Keras. tf.Keras doesn't seem to have the same information.